contrib/win32/install/install/components/c_server.wxs

```diff
@@ -0,0 +1,138 @@
+<Wix
+  xmlns="http://schemas.microsoft.com/wix/2006/wi"
+  xmlns:fw="http://schemas.microsoft.com/wix/FirewallExtension"
+  xmlns:util="http://schemas.microsoft.com/wix/UtilExtension">
+  <Fragment>
+    <ComponentGroup Id="CG_SERVER_FILES" Directory="INSTALLFOLDER">
+      <Component Id="cmp_sftpserver.exe">
+        <File Id="sftpserver.exe" KeyPath="yes" Source="$(var.SOURCEDIR)/sftp-server.exe" Name="sftp-server.exe" />
+      </Component>
+      <Component Id="cmp_sshshellhost.exe">
+        <File Id="sshshellhost.exe" KeyPath="yes" Source="$(var.SOURCEDIR)/ssh-shellhost.exe" Name="ssh-shellhost.exe" />
+      </Component>
+      <Component Id="cmp_opensshevents.man">
+        <File Id="opensshevents.man" KeyPath="yes" Source="$(var.SOURCEDIR)/openssh-events.man" Name="openssh-events.man">
+          <util:EventManifest />
+        </File>
+        <util:XmlFile
+          Id="xmlSetResourceFileName"
+          Sequence="1"
+          File="[#opensshevents.man]"
+          Action="setValue"
+          ElementPath="//*[\[]local-name() = 'provider'[\]]"
+          Name="resourceFileName"
+          Value="[#sshagent.exe]"
+          SelectionLanguage="XPath" />
+        <util:XmlFile
+          Id="xmlSetMessageFileName"
+          Sequence="2"
+          File="[#opensshevents.man]"
+          Action="setValue"
+          ElementPath="//*[\[]local-name() = 'provider'[\]]"
+          Name="messageFileName"
+          Value="[#sshagent.exe]"
+          SelectionLanguage="XPath" />
+      </Component>
+    </ComponentGroup>
+    <ComponentGroup Id="CG_SERVER_SSHAGENTSERVICE" Directory="INSTALLFOLDER">
+      <Component Id="cmp_sshagent.exe">
+        <File Id="sshagent.exe" Source="$(var.SOURCEDIR)/ssh-agent.exe" Name="ssh-agent.exe" KeyPath="yes" />
+        <ServiceInstall
+          Id="SVC_SSHAGENT"
+          Description="Agent to hold private keys used for public key authentication."
+          DisplayName="OpenSSH Authentication Agent"
+          ErrorControl="critical"
+          Interactive="no"
+          Name="ssh-agent"
+          Start="auto"
+          Type="ownProcess"
+          Vital="yes">
+          <util:ServiceConfig
+            ResetPeriodInDays="1"
+            FirstFailureActionType="restart"
+            SecondFailureActionType="restart"
+            ThirdFailureActionType="restart" />
+          <PermissionEx Sddl="D:(A;;CCLCSWRPWPDTLOCRRC;;;SY)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCLCSWLOCRRC;;;IU)(A;;CCLCSWLOCRRC;;;SU)(A;;RP;;;AU)" />
+        </ServiceInstall>
+        <ServiceControl
+          Id="SC_SSHAGENT_UNINSTALL"
+          Name="ssh-agent"
+          Stop="uninstall"
+          Remove="uninstall"
+          Wait="no" />
+      </Component>
+      <Component Id="cmp_server_sshagent_control" KeyPath="yes" Guid="5f300a9e-8896-4b3e-8b17-8ab1360b3ef1">
+        <Condition>
+          <![CDATA[NOT(DONT_START_SSHAGENT OR DONT_START_SERVICES)]]>
+        </Condition>
+        <ServiceControl
```

Comment:
@bagajjal @SteveL-MSFT any reason to conditionally start the service?

Reply:
Yes. sshd_config (sshd serer configuration) can be configured to be more secured based on the users need,
1) changing default port from 22 to user specific port to avoid security attacks.
2) An admin user can configure which users are allowed to login (Allowusers, DenyUsers)
3) An admin user can configure the type of access like providing only sftp access and denying interactive shell, etc.

Reply:
They can edit the file and restart the service immediately after install. Is putting a condition on this something you think people will actually use? Does it need to be exposed in the UI that most admins will probably use?

/cc @SteveL-MSFT

Reply:
As part of MSI, I would give an option (disabled by default, to match existing install-sshd.ps1 behavior) to start the sshd, ssh-agent service. This gives control for admin users who want to first make changes to sshd_config before starting the sshd service.